lib/engines/hcktest/tests.rb

```diff
@@ -208,7 +213,17 @@ def test_finished?(test)
     end
 
     def done_tests
-      @tests.select { |test| test_finished?(test) }
+      # When test is running more than ones HLK reports test['status'] = PASS/FAIL
+      # even when test is still running again. So we need to check test['executionstate']
+      # to be sure that test is really finished.
+      # Otherwise new_done function can report test as finished multiple times just with
+      # different executionstate.
+      @tests.select { |test| test_finished?(test) && test['executionstate'] == 'NotRunning' }
```

Comment:
I think this change is no longer necessary.

Reply:
we use `done_tests` for CI status. This change fixes displayed information.

Reply:
It will be inconsistent with `cnt_passed` and `cnt_failed` in `tests_stats`. Correct `cnt_passed` and `cnt_failed`, and derive `currentcount` by summing them.